networkx/linalg/laplacianmatrix.py

```diff
@@ -147,36 +147,115 @@ def normalized_laplacian_matrix(G, nodelist=None, weight="weight"):
 
     Returns
     -------
-    float
-        The sum of the total multiplicative weights for all spanning trees in `G`
+    t : float
+        Undirected graphs:
+            The sum of the total multiplicative weights for all spanning trees in `G`.
+        Directed graphs:
+            The sum of the total multiplicative weights for all spanning trees of `G`,
+            rooted at node `root`.
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If `G` does not contain any nodes.
+
+    NetworkXError
+        If the graph `G` is not (weakly) connected,
+        or if `G` is directed and the root node is not specified or not in G.
+
+    Examples
+    --------
+    >>> G = nx.complete_graph(5)
+    >>> round(nx.total_spanning_tree_weight(G))
+    125
+
+    >>> G = nx.Graph()
+    >>> G.add_edge(1, 2, weight=2)
+    >>> G.add_edge(1, 3, weight=1)
+    >>> G.add_edge(2, 3, weight=1)
+    >>> round(nx.total_spanning_tree_weight(G, "weight"))
+    5
+
+    Notes
+    -----
+    Self-loops are excluded. Multi-edges are contracted in one edge
+    equal to the sum of the weights.
+
+    References
+    ----------
+    .. [1] Wikipedia
+       "Kirchhoff's theorem."
+       https://en.wikipedia.org/wiki/Kirchhoff%27s_theorem
+    .. [2] Kirchhoff, G. R.
+        Über die Auflösung der Gleichungen, auf welche man
+        bei der Untersuchung der linearen Vertheilung
+        Galvanischer Ströme geführt wird
+        Annalen der Physik und Chemie, vol. 72, pp. 497-508, 1847.
+    .. [3] Margoliash, J.
+        "Matrix-Tree Theorem for Directed Graphs"
+        https://www.math.uchicago.edu/~may/VIGRE/VIGRE2010/REUPapers/Margoliash.pdf
     """
     import numpy as np
+    import scipy as sp
+
+    graph_is_directed = nx.is_directed(G)
+    if graph_is_directed == False and not nx.is_connected(G):
+        raise nx.NetworkXError("Graph G must be connected.")
+    if graph_is_directed and root == None:
+        raise nx.NetworkXError("Spanning trees in directed graphs require a root node.")
+    if graph_is_directed and root not in G:
+        raise nx.NetworkXError("The node root is not in the graph G.")
+    if graph_is_directed and not nx.is_weakly_connected(G):
+        raise nx.NetworkXError("Graph G must be weakly connected.")
+
+    # Compute directed Laplacian matrix
+    if graph_is_directed == False:
+        L = nx.laplacian_matrix(G, weight=weight)
+    else:
+        A = nx.adjacency_matrix(G, weight=weight)
+        out_deg = [d for a, d in G.out_degree(weight=weight)]
+        n, m = A.shape
+        # TODO: rm csr_array wrapper when spdiags can produce arrays
+        D = sp.sparse.csr_array(sp.sparse.spdiags(out_deg, 0, m, n, format="csr"))
+        L = D - A
+
+    # Remove one row/column corresponding to node `root`
+    if graph_is_directed == False:
+        i = 0
+    else:
+        i = list(G).index(root)
+    L = L.todense()
+    L = np.delete(L, i, 0)
+    L = np.delete(L, i, 1)
```

Comment:
Can you add `axis=` ?  Also a real nit-picky thing: switch the order of the if/else clause defining `i` so that you can use the `if graph_is_directed:` without any negation. Easier to read... ?maybe?